File: pop-miners/veriblock-pop-miner/src/main/kotlin/org/veriblock/miners/pop/storage/OperationStateData.java
// VeriBlock PoP Miner
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package org.veriblock.miners.pop.storage;

import com.j256.ormlite.field.DataType;
import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

@DatabaseTable(tableName = "operation_state")
public class OperationStateData {
    @DatabaseField(columnName = "id", id = true)
    public String id;
    @DatabaseField(columnName = "endorsed_block_number", index = true)
    public int endorsedBlockNumber;
    @DatabaseField(columnName = "endorsed_block_hash", index = true)
    public String endorsedBlockHash;
    @DatabaseField(columnName = "status")
    public String status;
    @DatabaseField(columnName = "action")
    public String action;
    @DatabaseField(columnName = "transaction_status", index = true)
    public String transactionStatus;
    @DatabaseField(columnName = "message")
    public String message;
    @DatabaseField(columnName = "state", dataType = DataType.BYTE_ARRAY)
    public byte[] state;
    @DatabaseField(columnName = "is_done")
    public boolean isDone;
    @DatabaseField(columnName = "last_updated", index = true)
    public int lastUpdated;
}

File: pop-miners/veriblock-pop-miner/src/main/kotlin/org/veriblock/miners/pop/storage/OperationStateData.kt
// VeriBlock PoP Miner
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package org.veriblock.miners.pop.storage

import com.j256.ormlite.field.DataType
import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable

@DatabaseTable(tableName = "operation_state")
class OperationStateData {
    @DatabaseField(columnName = "id", id = true)
    var id: String? = null

    @DatabaseField(columnName = "endorsed_block_number", index = true)
    var endorsedBlockNumber = 0

    @DatabaseField(columnName = "endorsed_block_hash", index = true)
    var endorsedBlockHash: String? = null

    @DatabaseField(columnName = "status")
    var status: String? = null

    @DatabaseField(columnName = "action")
    var action: String? = null

    @DatabaseField(columnName = "transaction_status", index = true)
    var transactionStatus: String? = null

    @DatabaseField(columnName = "message")
    var message: String? = null

    @DatabaseField(columnName = "state", dataType = DataType.BYTE_ARRAY)
    var state: ByteArray? = null

    @DatabaseField(columnName = "is_done")
    var isDone = false

    @DatabaseField(columnName = "last_updated", index = true)
    var lastUpdated = 0
}
